### PR TITLE
[cherry-pick](branch-2.0) support delete sub pred v2

### PR DESCRIPTION
**Pull request by Hastyshell**
## Proposed changes

pick #22442

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by yiguolei** [comment]
hold this PR, we should pick another PR before this PR.

**Comment by TangSiyang2001** [comment]
run buildall

File: be/src/olap/delete_handler.cpp [hunk 1]
```diff
@@ -71,6 +74,9 @@ Status DeleteHandler::generate_delete_predicate(const TabletSchema& schema,
     for (const TCondition& condition : conditions) {
         if (condition.condition_values.size() > 1) {
             InPredicatePB* in_pred = del_pred->add_in_predicates();
+            if (condition.__isset.column_unique_id) {
+                in_pred->set_column_unique_id(condition.column_unique_id);
```
Comment (yiguolei) [inline be/src/olap/delete_handler.cpp:78]: 这里的condition 的 unique id 不一定被设置了吧？ 如果没设置，此时是不是即使是v2 ，那么也没有column unique id？？

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by yiguolei** [comment]
Hold this PR， something is not very clear. We should check some code.

**Comment by xiaokang** [comment]
not merge this feature in 2.0

**Pull request closed without merging**